### PR TITLE
tools/sched_tp: Update sched_tp module to latest

### PR DESCRIPTION
**Pull request by mishradk**
update sched_tp module to include new events as provided
by Qais.Yousef@arm.com
removed 0001-sched-add-a-module-to-convert-tp-into-events.patch

### CODE REVIEW

File: tools/kmodules/sched_tp/Makefile [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
 EXTRA_CFLAGS = -I$(src)
+
+VMLINUX_DEPS_H = vmlinux_deps.h
+VMLINUX_H = vmlinux.h
+
+VMLINUX_DEPS_TXT = vmlinux_deps.txt
+VMLINUX_TXT = vmlinux.txt
+
+ifeq ($(wildcard $(KERNEL_SRC)/vmlinux), )
+	VMLINUX = /sys/kernel/btf/vmlinux
```
Comment (douglas-raillard-arm) [inline tools/kmodules/sched_tp/Makefile:12]: We should probably remove this case, LISA is expected to be installed on a host that is different from the target and everything cross compiled, so this Makefile will be called on the host machine, not the development board.
Therefore we can't use the host's kernel to infer anything about the kernel's target.

File: tools/kmodules/sched_tp/Makefile [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+
+install:
+	make -C $(KERNEL_SRC) M=$(PWD) modules_install
+
+clean:
+	make -C $(KERNEL_SRC) M=$(PWD) clean
+	rm -f $(VMLINUX_H) $(VMLINUX_DEPS_H)
+
+$(VMLINUX_DEPS_H): $(VMLINUX_DEPS_TXT) $(VMLINUX)
+	pahole -C file://vmlinux_deps.txt $(VMLINUX) > $@
```
Comment (douglas-raillard-arm) [inline tools/kmodules/sched_tp/Makefile:28]: Looks like `file://vmlinux_deps.txt` should be replaced with `$(VMLINUX_DEPS_TXT)`

File: tools/kmodules/sched_tp/Makefile [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+
+$(VMLINUX_H): $(VMLINUX_DEPS_H) $(VMLINUX_TXT) $(VMLINUX)
+	pahole -C file://vmlinux.txt $(VMLINUX) > $@
```
Comment (douglas-raillard-arm) [inline tools/kmodules/sched_tp/Makefile:31]: ditto for the hardcoded string

**Review by douglas-raillard-arm — CHANGES_REQUESTED** [review]

File: tools/kmodules/sched_tp/Makefile (hunk 1 above)
Comment (mishradk) [reply tools/kmodules/sched_tp/Makefile]: Updated the changed.

File: tools/kmodules/sched_tp/Makefile (hunk 2 above)
Comment (mishradk) [reply tools/kmodules/sched_tp/Makefile]: Modified the code.

File: tools/kmodules/sched_tp/Makefile (hunk 3 above)
Comment (mishradk) [reply tools/kmodules/sched_tp/Makefile]: Modified the code to use $(VMLINUX_TXT)

**Comment by douglas-raillard-arm** [comment]
LGTM, let's see how the pahole version issue is handled before merging

**Comment by douglas-raillard-arm** [comment]
Superseded by https://github.com/ARM-software/lisa/pull/1713

**Pull request closed without merging**